src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -207,6 +207,23 @@ public static TextSpan RangeToTextSpan(LSP.Range range, SourceText text)
             {
                 try
                 {
+                    // Per LSP spec, LSP.Range end position is exclusive. If you want to specify a range that 
+                    // includes the line ending characters then use an end position denoting the start of the next line.
+                    //
+                    // The TextLines will throw a ArgumentException if asked for the line after the last line,
+                    // despite the call being valid call per the LSP spec. If the end of the input range is one
+                    // beyond the end of the file, edit the end character position of the last line in the file.
+                    if (text.Lines.Count == linePositionSpan.End.Line)
+                    {
+                        var indexOfLastLine = text.Lines.Count - 1;
+                        var lastLineOfText = text.Lines[indexOfLastLine];
+                        // Number of characters in the line (inclusive of line ending characters) are the start to the end of line break
+                        var endLineCharacterPosition = lastLineOfText.EndIncludingLineBreak - lastLineOfText.Start;
+
+                        var editedEndLinePosition = new LinePosition(indexOfLastLine, endLineCharacterPosition);
+                        linePositionSpan = new LinePositionSpan(linePositionSpan.Start, editedEndLinePosition);
+                    }
```

Comment:
@dibarbet Just realized you recommended this be in RangeToLinePosition.

Reply:
Looks like it's not used (unless there is a specification I don't know about) so I've removed it.